libs/application/core/src/types/BasicDataProvider.ts

```diff
@@ -13,13 +14,32 @@ export interface DataProvider {
 
 export abstract class BasicDataProvider implements DataProvider {
   readonly type!: string
+  readonly accessToken: string
+
+  constructor(accessToken = '') {
+    this.accessToken = accessToken
+  }
 
   /**
    * Use this method to fetch data from external APIs
    * @param application: current application object which may or may not possess answers, and more information that
    * could be beneficial in the function body
    */
-  abstract provide(application: Application): Promise<unknown>
+  abstract async provide(application: Application): Promise<unknown>
+
+  protected async useGraphqlGateway(query: string): Promise<Response> {
+    return fetch(`http://localhost:4444/api/graphql`, {
```

Comment:
Hard coded url.

Reply:
Is it alright to read environment variables inside a lib?

Reply:
In a similar situation asked a developer to pass base url in as a param and then the app using the lib is using "Environment" resolvers for local and prod env.